### DISCUSSION TITLE
glTF Working Group Meeting 2021-10-28

### DISCUSSION THREAD

**Discussion by robertlong** [omigroup/gltf-extensions#30] [General]
# Agenda
- Ratification Process
  - Go over and potentially vote on whether or not to use 4 stage process
    - Still need to figure out the IPR bot...
    - https://github.com/omigroup/gltf-extensions/pull/23 
  - Go over and potentially vote on implementation matrix
- Update on Implementation Status
  - Spec
    - Need to add markdown explainer
    - Publish blog post advertising the OMI_audio_emitter ecosystem
  - omi-gltf-samples-repo
  - three-omi
    - Stable and up to date
    - Transitioned to typescript
  - Third Room
    - Support is finished and merged: https://github.com/thirdroom/thirdroom/pull/17
  - Unity
    - https://github.com/omigroup/gltf-extensions/discussions/11 
  - Godot
    - https://github.com/godotengine/godot-proposals/issues/3273
    - https://github.com/godotengine/godot/pull/53738
    - https://github.com/godotengine/godot/pull/52407
    - https://github.com/godotengine/godot/pull/52541
  - Blender
    - https://github.com/omigroup/gltf-extensions/discussions/26 
    - https://github.com/omigroup/omi-blender-gltf/pull/1
  - VRChat
    - Blocked on OMI Unity glTF Editor Import
  - Altspace
    - Blocked on OMI Unity glTF Editor Import
  - Neos
    - Looking into MP3 player support and accessing proper data on scene nodes
  - Hubs
    - PR submitted to Hubs repo https://github.com/mozilla/hubs/pull/4741  
  - glTF Validator
    - Up to date with the latest spec 
  - glTF Transform Tool
    - Up to date with the latest spec 
  - Stereo Kit
    - Waiting on an update from Nova

**Comment by robertlong** [comment]
# Notes
- Unity
  - Reach out to @atteneder to see what the status is for an import/export extension API in glTFast so we can use it.
  - In the meantime, perhaps we develop a hacked together solution based on parsing the JSON up front and adding extensions after the scene has been deserialized.
- Godot
  - Still waiting on Godot 4.0 alpha
- Blender
  - Basic implementation mostly finished
  - JSON being exported still need to work on bufferView/moving audio files
  - Planning on another hack day Tuesday evening
- Ratification Process
  - The IPR bot is fixed, you can leave comments on the PR here https://github.com/omigroup/gltf-extensions/pull/23
  - We made some changes to the language during this meeting and will be holding a vote at our next meeting on November 4th
  - We should add the existing resources we have for validation etc to the readme so that others can use it as a reference for writing new proposals.

**Reply by atteneder** [reply]
@robertlong Thanks for reaching out!

Work on extension API has not started and presumably will not start before end of the year. However, I'm already sketching things out I'm open to inputs and requirements. If you have concrete examples of extensions you'd like to inject or want to get involved in the API design, let me know.

Technical detail: Once the JSON deserialzation in glTFast was replaced with a more flexible (but still performant) solution, a custom extension API will get feasible. For the new export feature glTFast already uses a custom JSON serialization, so we're closer to something there.
